scripts/main/album.js

```diff
@@ -868,111 +961,150 @@ album.setProtectionPolicy = function (albumID) {
+		];
 
-	const dialogSetupCB = function () {
-		// TODO: If the modal dialog would provide this callback with proper jQuery objects for all input/select/choice elements, then we would not need these jQuery selectors
-		$('.basicModal .switch input[name="is_public"]').prop("checked", album.json.is_public);
-		$('.basicModal .switch input[name="is_nsfw"]').prop("checked", album.json.is_nsfw);
 		if (album.json.is_public) {
-			$(".basicModal .choice input").attr("disabled", false);
-			// Initialize options based on album settings.
-			$('.basicModal .choice input[name="grants_full_photo"]').prop("checked", album.json.grants_full_photo);
-			$('.basicModal .choice input[name="requires_link"]').prop("checked", album.json.requires_link);
-			$('.basicModal .choice input[name="is_downloadable"]').prop("checked", album.json.is_downloadable);
-			$('.basicModal .choice input[name="is_share_button_visible"]').prop("checked", album.json.is_share_button_visible);
-			$('.basicModal .choice input[name="has_password"]').prop("checked", album.json.has_password);
+			tristateCheckboxes.forEach(function (checkbox) {
+				checkbox.parentElement.classList.remove("disabled");
+				checkbox.disabled = false;
+			});
+			// Initialize options based on global settings.
+			formElements.grants_full_photo.checked = album.json.grants_full_photo;
+			formElements.requires_link.checked = album.json.requires_link;
+			formElements.is_downloadable.checked = album.json.is_downloadable;
+			formElements.is_share_button_visible.checked = album.json.is_share_button_visible;
+			formElements.has_password.checked = album.json.has_password;
 			if (album.json.has_password) {
-				$('.basicModal .choice input[name="passwordtext"]').show();
+				formElements.password.parentElement.classList.remove("hidden");
+			} else {
+				formElements.password.parentElement.classList.add("hidden");
 			}
 		} else {
-			$(".basicModal .choice input").attr("disabled", true);
+			tristateCheckboxes.forEach(function (checkbox) {
+				checkbox.parentElement.classList.add("disabled");
+				checkbox.disabled = true;
+			});
 			// Initialize options based on global settings.
-			$('.basicModal .choice input[name="grants_full_photo"]').prop("checked", lychee.grants_full_photo);
-			$('.basicModal .choice input[name="requires_link"]').prop("checked", false);
-			$('.basicModal .choice input[name="is_downloadable"]').prop("checked", lychee.is_downloadable);
-			$('.basicModal .choice input[name="is_share_button_visible"]').prop("checked", lychee.is_share_button_visible);
-			$('.basicModal .choice input[name="has_password"]').prop("checked", false);
-			$('.basicModal .choice input[name="passwordtext"]').hide();
+			formElements.grants_full_photo.checked = lychee.grants_full_photo;
+			formElements.requires_link.checked = false;
+			formElements.is_downloadable.checked = lychee.is_downloadable;
+			formElements.is_share_button_visible.checked = lychee.is_share_button_visible;
```

Comment:
> I wish we were getting a warning about such cases from some code checker though...

That's future work. Something like this is possible. A proper IDE and CI tools would be able to catch those things, but only if we would use proper JS modules and `use "strict";`. Without that the code above is actually not wrong. I am aiming into that direction, but it will still take some time.